packages/core/src/prisma-utils.ts

```diff
@@ -1,7 +1,7 @@
 import {exec} from "npm-run"
 
 interface Constructor<T = unknown> {
-  new (...args: unknown[]): T
+  new (...args: any[]): T
```

Comment:
Ouch, I botched it. I wanted to write "any possible constructor", and instead I wrote "any constructor that doesn't care about its arguments". 

Could we use `never` here? `any` is an escape hatch, and it tends to "leak". In "fully strict" TypeScript you'd have `never` and `unknown` in all places where `any` is commonly used. (unknown is top type, never is bottom type, any is "shut up TypeScript, I ain't got time"). It's a bit tricky — I had to pull the PR and tested it to make sure. 

[TS Playground with an example](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIHsQGcxQK4JjpQBiU6AtgHLpgAWoA5gDwAqyAvMniANYjoA7iAB8yAN4AoZMhARByABQA6VXCiMsALlkQAbtADaAXQCUO1pIC+kyaEixEKDNlwEipchQDqdOJAMoNk5uPgFhMSkZOQUVNQ1tUP4hEBNzZEsbO3BoeCQ0TBx8QmJvYHoAUSwEOAAHCAAJfwQ6YK4eZIiJaV1Y1WV1TR04EABPNItrWwQAGzgsLGQGvAoR7pkZBEK3EqhFWrwAIxngBFk4CggdIqZTCRsssFH6pZWRlyL3YhCn+vQYV6rEAAblsAHowcg3BBJL8UAB9ACMIWWQI+Ow8yAgAA9ICAACaLdHFDxkSg0ehMZAAfih+BQOngMywEFBkghyCZLNhzwRACYUW8QMSvlAsbiIASidsScQyT4-AFoDS6XgGZy4MzWeDIdCeS94QBmQVomWi8V4wkFVyyqBlSrVOqNZp0FXQ5CMzUs4FAA)

```suggestion
  new (...args: never[]): T
```


---

BTW I assumed that `strict: false` is here in examples tsconfigs because of Prisma codegen.
What do you think about adding strict typecheck to catch problems like this?

Reply:
ok sweet, thanks!

Reply:
I'll see about making auth example work with strict mode